src/types/asset.d.ts

```diff
@@ -0,0 +1,6 @@
+export interface IAsset {
+  toString(): string;
+  connect<T>(filters: object): Promise<T>;
+}
```

Comment:
I'm not sure that we need an interface for every single class to implement, especially since some of them are empty. We should only need interfaces where we have multiple classes that implement them or if we need an abstract interface that arbitrary objects (not class instances) must adhere to. Any thoughts on that?

Reply:
hey, sorry but I'm not sure if i completely understand this problem. if you could provide me an example of such more scenarios that would be really helpful, to get me an idea on what exactly needs to be done. as i haven't spent sufficient time on this codebase a kickstart would be good to me;